File: code/83 Remove Duplicates from Sorted List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteDuplicates(struct ListNode* head) {
    if(head){
        struct ListNode *p=head;
        while(p->next){
            if(p->val != p->next->val){
                p=p->next;
            }else{
                struct ListNode* temp=p->next;
                p->next=p->next->next;
                free(temp);
            }
        }
    }
    return head;
}

File: code/83 Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        prev_node = head
        if not prev_node: return None
        node = prev_node.next
        while node:
            if node.val == prev_node.val: prev_node.next = node.next
            else: prev_node = node
            node = node.next
        return head
